cmd/launcher/launcher.go

```diff
@@ -258,6 +258,8 @@ func runLauncher(ctx context.Context, cancel func(), slogger, systemSlogger *mul
 	signalListener := newSignalListener(sigChannel, cancel, logger)
 	runGroup.Add("sigChannel", signalListener.Execute, signalListener.Interrupt)
 
+	agent.ResetDatabaseIfNeeded(ctx, k)
```

Comment:
should we call this something else since it's not actually resetting anymore, or maybe leave a comment here?

Reply:
Oh yes, thank you, very good point! I will update

Reply:
Renamed and added a comment as well